### DISCUSSION TITLE
Support your project / New function suggestion

### DISCUSSION THREAD

**Discussion by user128160** [sk-zk/lookaround-map#7] [General]
Dear sk-zk,

your project is absolutely amazing. I have tried so many ways to get access to Apple Look Around on Windows including installing macOS on a virtual machine, but nothing worked out until I found your project. Thank you, a million times!

I am very sorry if my way of using it did cause any trouble to you, this was never my intention. Even as I tried before already, I couldn’t get your suggested solution working which you mentioned to me in your personal server message. I am sure it is super easy for Phyton experts, unfortunately I am still quite new with that but will keep trying to finally make it work.

Please let me know if I can donate you some money to compensate for possible exceeded bandwidth usage? I am very sorry again, I will not do it again.

Could I suggest you add a full-screen or screenshot option without the menu overlay? At least for me, this would be a super useful function.

Regarding the top and bottom projection issue, I am also trying to find a solution and for sure would share it with you if can make any progress. It seems the top and bottom images are somehow distorted and can’t really fit even if being placed in the correct position either in your experimental render view and as well in Photoshop 3D view.

As you already mentioned in your documentation, Apple Look Around usually does not offer historical images. However, you managed to add the option to choose older photos from the drop-down menu. In some places only one single old photo remaining, but the rest from the same photo set/tour seems unavailable or deleted. Do you think the other old images are just removed from the index, but sill somehow uploaded on the server and could be accessed with a different approach? Maybe from a single remaining photo ID, the whole photo set/tour could still be found and recovered? I saved some screenshots on my iPad before, so I know for sure that the other old images were existing in the past.

Is there any way to contact you directly in private message, maybe by email? As you probably already know, am from Germany as well and would be very happy to support you with this project. For sure German would be fine too.

Thank you very much!

**Comment by sk-zk** [comment]
Hey there

PythonAnywhere conveniently doesn't charge extra for traffic (that's why I went with them) - the problem was simply that getting a pano face is a very CPU-heavy operation due to the need to convert the image, so it was slowing down the site for other users. (Btw, if you don't need the conversion step, you can append `?heic=1` and the server will just relay the image unmodified.)

>I have tried so many ways to get access to Apple Look Around on Windows including installing macOS on a virtual machine

Yeah, it's tricky to set up because Apple Maps requires a GPU that MacOS has support for passed through to the VM. I eventually got it working with an RX480 using [OSX-KVM](https://github.com/kholia/OSX-KVM) on Ubuntu.

>I couldn’t get your suggested solution working which you mentioned to me in your personal server message

Make sure that [git](https://git-scm.com/download/win), [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) and [Python](https://www.python.org/downloads/) are installed, and run the commands in the Setup section in order. You should now have a local server which functions the same as the public one. Let me know if you're getting any error messages

>Could I suggest you add a full-screen or screenshot option without the menu overlay? At least for me, this would be a super useful function.

Sure, I can give that a go

>Do you think the other old images are just removed from the index, but sill somehow uploaded on the server and could be accessed with a different approach?

When new imagery is published, the metadata of the previous panoramas is no longer available, but the images themselves can still be accessed for a while if you have the IDs. I don't know for how long they stick around, but a year seems plausible. Here's a pano in Kiel from 2020 which was replaced with 2022 coverage in May 2023 but is still accessible:

<img src="https://i.imgur.com/ViFaCOW.png" width=400>

Since there's no way that I know of to fetch the metadata, though, you will need to determine the position and orientation of each panorama manually if you didn't cache that data as well.

>Maybe from a single remaining photo ID, the whole photo set/tour could still be found and recovered?

Yep! A pano ID in Look Around is a large, probably arbitrary number per drive plus the time since the start of the drive (or the current UNIX timestamp) in milliseconds. This means that you can simply keep incrementing (or decrementing) the ID until you've found the next (or previous) image in the sequence. But again, you won't have the metadata, so you will need to set position and orientation by hand.

>Is there any way to contact you directly in private message, maybe by email? 

Am einfachsten erreichst du mich über Discord: skzk_

**Comment by pm1974** [comment]
Hi sk-zk,

Merry Christmas and thank you very much for your detailed reply and your help!

I am really sorry that I didn’t get back to you any earlier. The past weeks were crazy and I didn’t have time at all to explore the Look Around / Street View topic any further.

But for sure I noticed that you already added the screenshot function a while ago. Thank you for that.

I was not aware that the image conversion requires so much CPU bandwidth. I understand that my previous download attempts were a problem and of course I will not do that again. I was trying to back up some images which I was afraid they might get deleted soon, but I never intended to harm your project or slow down the server. I am sorry again for the troubles I caused.

The additional command to download .heic images instead is great and also helps to save some disk space. Sure, this file format is not so much convenient in Windows but still workable with a few extensions and might become more popular and compatible in the future.

I just realized that you also already managed to fix the bottom face projection. That’s an absolute awesome improvement, congratulations!

Your research about the look around data base is really great. I wish Apple would decide to offer historical images as additional function instead of deleting them, but I guess there is no hope for that.

Discord is completely new for me but I already signed up and will contact you there.

Thank you again.
